application/configuration/types.go

```diff
@@ -117,15 +118,15 @@ type telegramBot struct {
 	ChatID int64
 }
 
-func newTelegramBot() telegramBot {
-	return telegramBot{
+func newTelegramBot() *telegramBot {
+	return &telegramBot{
 		Token:  "",
 		ChatID: 0,
 	}
 }
 
-func newKafkaConfig() kafkaConfig {
-	return kafkaConfig{
+func newKafkaConfig() *kafkaConfig {
```

Comment:
It won't protect us. Golang doesn't have const types at all:
1. Even on type by the value it is possible to perform a call of a method with the receiver by pointer and change values
2. All the configs include pointer types: slices, maps, interfaces. This value can be changed even having a type by value.

If we'd like to have immutability, then the only solution is to have all fields private with a bunch of getters/setters as methods + write their own serialization because private fields would break serialization, deserialization and reflection.

Unfortunately, there is no easy solution except having pointers and good test coverage with automated linters and CI.